android/src/com/mapswithme/maps/routing/NavigationController.java

```diff
@@ -112,11 +104,20 @@ public NavigationController(AppCompatActivity activity, @NonNull MapButtonsContr
 
     mStreetFrame = topFrame.findViewById(R.id.street_frame);
     mNextStreet = mStreetFrame.findViewById(R.id.street);
-    View shadow = topFrame.findViewById(R.id.shadow_top);
-    UiUtils.hide(shadow);
 
-    UiUtils.extendViewWithStatusBar(mStreetFrame);
-    UiUtils.extendViewMarginWithStatusBar(turnFrame);
+    // Show a blank view below the navbar to hide the menu content
```

Comment:
Sometimes, when Android 5 (and others) switch from a light to a dark theme on activating navigation mode, I see some white backgrounds at the top and bottom of the screen before correctly themed and drawn elements appear. Is it caused by this hack? Why exactly is it needed? Can it be avoided?

Reply:
I can't reproduce this issue but as the hack adds a bar tinted with the background color below the navigation bar, if the theme takes time to switch when going into navigation mode that bar will appear white.

This hack is needed to hide the navigation bottom sheet content when below the navigation bar. As the navigation bar is set to translucent, we cannot control its color. Adding a view behind it was the simplest way I found to make it temporarily opaque.